website_multi_company/hooks.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, SUPERUSER_ID
+
+
+def post_init_hook(cr, _):
+
+    with api.Environment.manage():
+
+        env = api.Environment(cr, SUPERUSER_ID, {})
+        websites = env['website'].search([])
+        public_user = env.ref('base.public_user')
+        used_existing = False
+
+        for website in websites:
+
+            if all((
+                website.company_id == public_user.company_id,
```

Comment:
Isn't this line redundant?

Reply:
Redundant with which other line?

Reply:
If the following one. If you have already sync `company_id` with the company of the user, you don't need this check.

Reply:
This is covering the normal case of a 1:1 `website`:`public_user` relationship. In this case, no user is created and no synchronization occurs. 

The `used_existing` part is just in case there are 2 websites assigned to the 1 public user. This would happen if someone maybe created a website for use in `website_multi_theme`, then decided to install this one to add the company isolation.

In all cases other than the standard 1 website to 1 company to 1 user, a new public user is created and assigned to the website.

There is one edge case in this logic that could cause the default public user to be assigned to the default website, which would be if the company in either has been desynchronized manually for some reason.